### PR TITLE
feat(_spinner,avatar_change,avatar-mixin):adds feature for loading…

### PR DESCRIPTION
**Pull request by himanish-star**
Fixes https://github.com/mozilla/fxa-content-server/issues/5006

… spinner of avatars

**changes made are as follows**
simplescreenrecorder-2017-12-22_11 43 23

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: b87406e91b94b13ecf2cad3860f288034c423c8f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 4499211314db47038525da778156d0c6eaaa05d2
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by himanish-star** [comment]
@shane-tomlinson , @ryanfeeley , I was going through the logs of the travis CI build and found out that the command grunt nsp failed three times

        The command "grunt nsp" failed 3 times.
        The command "travis_retry grunt nsp" exited with 6.

But then I tried running locally 

    grunt nsp
    Running "nsp" task
    (node:22723) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
    (+) No known vulnerabilities found
    Done.


    Execution Time (2017-12-22 14:26:15 UTC+5:30)
    loading tasks  5.7s  
    ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 48%
    nsp            6.1s  
    ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 52%
    Total 11.7s

This clears the test . So what do I do now

**Comment by vladikoff** [comment]
@himanish-star when pressing "Add" avatar with no avatar set I get an empty avatar and a spinner, I guess there should be no spinner in that case: 

image

File: app/scripts/templates/settings/avatar_change.mustache [hunk 1]
```diff
@@ -1,7 +1,10 @@
...
+      </div>
+      <p style="display: none" id="image-holder" class="avatar-wrapper">
```
Comment (vladikoff) [inline app/scripts/templates/settings/avatar_change.mustache]: do not add inline styles such as `style="display: none"`

File: app/scripts/views/mixins/avatar-mixin.js [hunk 2]
```diff
@@ -87,6 +87,8 @@ define(function (require, exports, module) {
             avatarWrapperEl
               .removeClass('with-default')
               .append($(profileImage.get('img')).addClass('profile-image'));
+            $('#image-holder').css('display','inline-block');
```
Comment (vladikoff) [inline app/scripts/views/mixins/avatar-mixin.js]: add a space after `,` in both cases `.css('display','inline-block');` -> `.css('display', 'inline-block');

File: app/styles/modules/_spinner.scss [hunk 3]
```diff
@@ -9,6 +9,19 @@
   width: 36px;
 }
 
+#loading-avatar-spinner img{
```
Comment (vladikoff) [inline app/styles/modules/_spinner.scss]: `#loading-avatar-spinner img{` ---> `#loading-avatar-spinner img {`

**Comment by himanish-star** [comment]
@vladikoff , I'll surely do all the changes requested. But I need your help. My master branch is exactly the same as upstream's master branch and when I run npm start my server starts on localhost:3030. But the spinner for loading keeps spinning for ages and I get this error when I open up the browser console
image

image

File: app/styles/modules/_spinner.scss (hunk 3 above)
Comment (himanish-star) [reply app/styles/modules/_spinner.scss]: Ok

File: app/scripts/views/mixins/avatar-mixin.js (hunk 2 above)
Comment (himanish-star) [reply app/scripts/views/mixins/avatar-mixin.js]: OK

File: app/scripts/templates/settings/avatar_change.mustache (hunk 1 above)
Comment (himanish-star) [reply app/scripts/templates/settings/avatar_change.mustache]: Ok

**Comment by vladikoff** [comment]
@himanish-star are you using `npm run start-remote`?

**Comment by vladikoff** [comment]
@himanish-star in your `local.json` under `server/config/` , try flipping `csp` to `false` and see if that loads

Ref: https://github.com/mozilla/fxa-content-server/blob/master/server/config/local.json-dist#L26

**Comment by himanish-star** [comment]
@vladikoff , extremely sorry. It is working now. `npm run start-remote` had just slipped of my mind. I'll soon fix the changes requested

**Comment by himanish-star** [comment]
@vladikoff, I have made the requested changes. 
> @himanish-star when pressing "Add" avatar with no avatar set I get an empty avatar and a spinner, I guess there should be no spinner in that case:

This is happening maybe because your browser has cached the data. But if you do a hard reload everything works. If you want I can make a video  and send it to over here to show how the things work out

**Comment by himanish-star** [comment]
Here is the GIF to show the changes
ezgif com-video-to-gif 10

**Comment by himanish-star** [comment]
@vbudhram , could you review my PR ? :smiley:

**Comment by himanish-star** [comment]
@vladikoff , I have made the changes please see to them

**Comment by himanish-star** [comment]
@vladikoff, any updates?

**Comment by vladikoff** [comment]
Landed in https://github.com/mozilla/fxa-content-server/commit/6bd8b5cd4473c45e19f277f437c6e380ddf80f7e

Thank you!!!

**Comment by himanish-star** [comment]
@vladikoff, also I'll update all my work in this repo by this weekend. Sorry for the delay and thanks

**Pull request closed without merging**